server/services/memory/migration_task.py

```
@@ -0,0 +1,157 @@
+from celery_worker import celery
+from pypdf import PdfReader
+from utils.db.connection_pool import db_pool
+from utils.storage.storage import get_storage_manager
+from services.artifacts.store import create_version
+from services.memory.splitter import split_content, make_part_title, make_part_description
+
+logger = logging.getLogger(__name__)
+
+
+@celery.task(name="migrate_kb_to_memory", bind=True, max_retries=0)
+def migrate_kb_to_memory(self):
```

Ruff (0.15.18)

[warning] Too many branches (17 > 12)

(PLR0912)